Mathlib/GroupTheory/MonoidLocalization.lean

```diff
@@ -1609,6 +1609,27 @@ theorem of_mulEquivOfMulEquiv {k : LocalizationMap T Q} {j : M ≃* P} (H : S.ma
 #align submonoid.localization_map.of_mul_equiv_of_mul_equiv Submonoid.LocalizationMap.of_mulEquivOfMulEquiv
 #align add_submonoid.localization_map.of_add_equiv_of_add_equiv AddSubmonoid.LocalizationMap.of_addEquivOfAddEquiv
 
+
+@[to_additive]
+theorem injective_iff (f : LocalizationMap S N) :
+    Injective (LocalizationMap.toMap f) ↔ ∀ ⦃x⦄, x ∈ S → IsLeftRegular x := by
+  rw [Injective]
+  constructor <;> intro h
+  · intro x hx
+    rw [IsLeftRegular,Injective]
+    simp_rw [LocalizationMap.eq_iff_exists] at h
+    intro y z hyz
+    apply (fun y z _ => h) y z x
+    have : ∃(x' : S), x' = x := (CanLift.prf x hx)
+    obtain ⟨x',hx'⟩ := this
+    rw [← hx'] at hyz
```

Comment:
The reason `CanLift` even exists is that there is a corresponding tactic that does all this for you!
```suggestion
    lift x to S using hx
```